include/irr/asset/IImageLoader.h

```diff
@@ -19,6 +19,52 @@ namespace irr
 				IImageLoader() = default;
 				virtual ~IImageLoader() = 0;
 
+				//! A function converting data in EF_R8_SRGB format to EF_R8G8B8_SRGB
+				/**
+					\bredChannelDataLayer\b parameter is consider as bunch of texels 
+					for certain single layer in region.
+					Region parameter is used to provide flexibility with data texel
+					convertion for various mipmap levels.
+
+					The reson for existance of the function is that R8 format with SRGB encoding used to be available in 
+					compatibility (with 2.1) OpenGL profile, but got removed from 3.3 core profile only to be added back to 
+					OpenGL 4.6, so not many GPU's drivers can natively accept and display a data, so the convertion is available.
+				*/
+
+				uint8_t* convertR8SRGBdataIntoRGB8SRGBAAndGetIt(const void* redChannelDataLayer, const core::smart_refctd_ptr<ICPUImage>& image, const irr::asset::IImage::SBufferCopy& region)
+				{
+					constexpr auto channelAmount = 3;
+					const void* planarData[] = { redChannelDataLayer, nullptr, nullptr, nullptr };
+					uint8_t* out = nullptr;
+					const size_t texelAmount = getFormatChannelCount(image->getCreationParameters().format);
+
+					const auto& imageSize = image->getCreationParameters().extent;
+					irr::core::vector3d<uint32_t> imageSizeWithPitch;
+					imageSizeWithPitch.X = region.bufferRowLength > 0 ? region.bufferRowLength : region.imageExtent.width;
+					imageSizeWithPitch.Y = region.bufferImageHeight > 0 ? region.bufferImageHeight : region.imageExtent.height;
+					imageSizeWithPitch.Z = region.imageExtent.depth;
+
+					video::convertColor<EF_R8_SRGB, EF_R8G8B8_SRGB>(planarData, out, texelAmount, imageSizeWithPitch);
```

Comment:
yeah I know, but I thought that it will be easy to expand the buffer in that way
but I will just get the size and multiple through 3 including all you mentioned

Reply:
Actually I thought about it as I will get from convert Color a ready to use pointer with appropriate size, you're right

Reply:
have to read about `Discrete Cosine Transform`

Reply:
well it doesnt matter what DCT is, its just that some image formats will require physical images to be multiples of some sort of a size because of compression or encoding, so padding will be added for the "unnatural" image resolutions.

Example HEVC or h.264 requires block sizes of 8x8 or something like that so you cant have a 400x300 video